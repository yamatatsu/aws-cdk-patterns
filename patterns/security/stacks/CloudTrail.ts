import * as cdk from "@aws-cdk/core"
import * as logs from "@aws-cdk/aws-logs"
import * as cloudtrail from "@aws-cdk/aws-cloudtrail"

export class CloudTrail extends cdk.Stack {
  public readonly logGroup: logs.ILogGroup

  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    const logGroup = new logs.LogGroup(this, "CloudTrailLogGroup", {
      retention: logs.RetentionDays.THREE_MONTHS,
    })

    new cloudtrail.Trail(this, "CloudTrail", {
      includeGlobalServiceEvents: true,
      isMultiRegionTrail: true,
      enableFileValidation: true,
      managementEvents: cloudtrail.ReadWriteType.ALL,
      sendToCloudWatchLogs: true,
      cloudWatchLogGroup: logGroup,
    })

    this.logGroup = logGroup
  }
}
