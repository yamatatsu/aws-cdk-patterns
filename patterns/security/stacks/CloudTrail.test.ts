import { SynthUtils } from "@aws-cdk/assert"
import * as cdk from "@aws-cdk/core"
import { CloudTrail } from "./CloudTrail"

test("snapshot test", () => {
  const app = new cdk.App()

  const target = new CloudTrail(app, "Target")

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot()
})
