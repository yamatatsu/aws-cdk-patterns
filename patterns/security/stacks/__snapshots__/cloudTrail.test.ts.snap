// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Resources": Object {
    "CloudTrailA62D711D": Object {
      "DependsOn": Array [
        "CloudTrailLogsRoleDefaultPolicyD9019B15",
        "CloudTrailLogsRole9F6E6663",
        "CloudTrailS3PolicyEA49A03E",
      ],
      "Properties": Object {
        "CloudWatchLogsLogGroupArn": Object {
          "Fn::GetAtt": Array [
            "CloudTrailLogGroup343A29D6",
            "Arn",
          ],
        },
        "CloudWatchLogsRoleArn": Object {
          "Fn::GetAtt": Array [
            "CloudTrailLogsRole9F6E6663",
            "Arn",
          ],
        },
        "EnableLogFileValidation": true,
        "EventSelectors": Array [
          Object {
            "IncludeManagementEvents": true,
            "ReadWriteType": "All",
          },
        ],
        "IncludeGlobalServiceEvents": true,
        "IsLogging": true,
        "IsMultiRegionTrail": true,
        "S3BucketName": Object {
          "Ref": "CloudTrailS310CD22F2",
        },
      },
      "Type": "AWS::CloudTrail::Trail",
    },
    "CloudTrailLogGroup343A29D6": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "RetentionInDays": 90,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "CloudTrailLogsRole9F6E6663": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cloudtrail.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CloudTrailLogsRoleDefaultPolicyD9019B15": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutLogEvents",
                "logs:CreateLogStream",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "CloudTrailLogGroup343A29D6",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CloudTrailLogsRoleDefaultPolicyD9019B15",
        "Roles": Array [
          Object {
            "Ref": "CloudTrailLogsRole9F6E6663",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CloudTrailS310CD22F2": Object {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "CloudTrailS3PolicyEA49A03E": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "CloudTrailS310CD22F2",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetBucketAcl",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cloudtrail.amazonaws.com",
              },
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "CloudTrailS310CD22F2",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "s3:PutObject",
              "Condition": Object {
                "StringEquals": Object {
                  "s3:x-amz-acl": "bucket-owner-full-control",
                },
              },
              "Effect": "Allow",
              "Principal": Object {
                "Service": "cloudtrail.amazonaws.com",
              },
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "CloudTrailS310CD22F2",
                        "Arn",
                      ],
                    },
                    "/AWSLogs/",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;
